test/server/drain_manager_impl_test.cc

```diff
@@ -37,12 +40,15 @@ TEST(DrainManagerImplTest, All) {
   // Test drain sequence.
   Event::MockTimer* drain_timer = new Event::MockTimer(&server.dispatcher_);
```

Comment:
I think it would be good to document in the style guide since this explicitly departs from the usual unique_ptr hygiene.  Agree it makes total sense for test code once I got the explanation.